File: app/src/main/java/com/prey/actions/aware/AlarmAwareReceiver.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2024 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.aware;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.prey.PreyLogger;

public class AlarmAwareReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        try {
            final Context ctx = context;
            new Thread() {
                public void run() {
                    new AwareController().init(ctx);
                }
            }.start();
        } catch (Exception e) {
            PreyLogger.e(String.format("error:%s", e.getMessage()), e);
        }
    }
}

File: app/src/main/java/com/prey/actions/aware/AlarmAwareReceiver.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.aware

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.prey.PreyLogger
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

/**
 * AlarmAwareReceiver is a BroadcastReceiver that listens for alarm events and initializes the AwareController.
 */
class AlarmAwareReceiver : BroadcastReceiver() {
    /**
     * Called when the BroadcastReceiver receives an Intent.
     *
     * @param context The Context in which the receiver is running.
     * @param intent The Intent being received.
     */
    override fun onReceive(context: Context, intent: Intent) {
        try {
            // Start a new thread to initialize the AwareController
            CoroutineScope(Dispatchers.IO).launch {
                // Initialize the AwareController with the application context
                AwareController().initLastLocation(context.applicationContext)
                AwareController.getInstance().registerGeofence(context)
            }
        } catch (e: Exception) {
            // Log any exceptions that occur during the onReceive method
            PreyLogger.e("AlarmAwareReceiver error: ${e.message}", e)
        }
    }

}